browser/src/app/Log.ts

```diff
@@ -0,0 +1,109 @@
+/* -*- js-indent-level: 8 -*- */
+/*
+ * Copyright the Collabora Online contributors.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * Logger contains methods for logging the activity
+ */
+
+type Direction = 'INCOMING' | 'OUTGOING';
+interface LogMsg {
+	msg: string;
+	direction: Direction;
+	time: number;
+}
+
+class Logger {
+	private _logs: LogMsg[];
+	private startTime: number;
+
+	constructor() {
+		this._logs = [];
+		this.startTime = null;
+	}
+
+	public log(msg: string, direction: Direction): void {
+		var time = Date.now();
+		if (!this.startTime) this.startTime = time;
+
+		// Limit memory usage of log by only keeping the latest entries
+		var maxEntries = 100;
+		if ((window as any).enableDebug) maxEntries = 1000;
+
+		if (time - this.startTime < 60 * 1000 /* ms */) maxEntries = 500; // enough to capture early start.
+		while (this._logs.length > maxEntries) this._logs.shift();
+
+		// Limit memory usage of log by limiting length of message
+		var maxMsgLen = 128;
+		if (msg.length > maxMsgLen) msg = msg.substring(0, maxMsgLen);
+		msg = msg.replace(/(\r\n|\n|\r)/gm, ' ');
+		this._logs.push({ msg: msg, direction: direction, time: time });
+	}
+
+	private _getEntries(): string {
+		this._logs.sort(function (a, b) {
+			if (a.time < b.time) {
+				return -1;
+			}
+			if (a.time > b.time) {
+				return 1;
+			}
+			return 0;
+		});
+		let data: string = '';
+		for (var i = 0; i < this._logs.length; i++) {
```

Comment:
```suggestion
		for (let i = 0; i < this._logs.length; i++) {
```

Let's stop using `var`

Reply:
done